nrf52-hal-common/src/uarte.rs

```diff
@@ -96,75 +121,44 @@ impl<T> Uarte<T> where T: UarteExt {
             w.baudrate().variant(baudrate)
         );
 
-        Uarte(uarte)
-    }
+        let (prod, cons) = MUH_BUFFAH.split();
 
-    /// Write via UARTE
-    ///
-    /// This method uses transmits all bytes in `tx_buffer`
-    ///
-    /// The buffer must have a length of at most 255 bytes
-    pub fn write(&mut self,
-        tx_buffer  : &[u8],
-    )
-        -> Result<(), Error>
-    {
-        // This is overly restrictive. See (similar SPIM issue):
-        // https://github.com/nrf-rs/nrf52/issues/17
-        if tx_buffer.len() > u8::max_value() as usize {
-            return Err(Error::TxBufferTooLong);
-        }
+        unsafe {
+            MAYBE_CONSUMER = Some(cons)
```

Comment:
Why safe the consumer in a `static` instead of returning it to the user?

Reply:
The current implementation of `Uarte` only supports sending data over serial. In this instance, the Producer is the application code, sending data, and the Consumer is used by the DMA/interrupt, to feed the outgoing data.